src/classes/HH_OppContactRoles_TDTM.cls

```diff
@@ -258,7 +258,7 @@ public class HH_OppContactRoles_TDTM extends TDTM_Runnable {
                                     thisRole.isPrimary = true;
                                     contactRolesForCreation.add(thisRole);
                                 }
-                            } else {
+                            } else if (!householdMember.Deceased__c) {
                                 // exclude any Contacts with an excluded record type
                                 if (recordTypesOnCons && conRecordTypesToExclude.contains((id)householdMember.get('RecordTypeId'))) {
```

Comment:
Nesting too many `IF` statements affects readability and makes maintenance more difficult. Consider refactoring to simplify the flow of this code block.